### ISSUE TITLE
Set non-mock module (with dependencies) in component

### ISSUE THREAD

**Issue by danielocampo2** [fabioCollini/DaggerMock#25]
Hi, I am trying to integrate DaggerMock in our project but I'm having troubles with our AndroidModule that has a dependency on another object (application).
This results in a runtime error when trying to build the ApplicationComponent, as the AndroidModule has to be set before hand.

My intention is to provide to DaggerMock the instance of this AndroidModule somehow. So far the only options I see are for DaggerMock to actually mock this module, but this is not my intention at all as I need this module to behave the normal way.

Any help is appreciated.
Thanks.

**Comment by fabioCollini** [comment]
Hi, are you using JVM or espresso tests? In both cases you can manually set the module when you create the rule. Here https://github.com/fabioCollini/DaggerMock/tree/master/RealWorldApp you can find a complete example.

**Comment by danielocampo2** [comment]
Hi @fabioCollini , thanks for your answer. I am using Espresso tests.

As far as I understand, the module that you set when you create the rule is the one that you want to mock, right?
My case is as follows: I want to mock ModuleA, so I pass a ModuleA instance to the rule. But to create the graph, there's another module, say ModuleX, that needs to be manually created (since it has a dependency), but I don't want this to be mocked.
I tried passing both modules, but it fails as ModuleX cannot be mocked. Same happens using the addComponentDependency method.

Let me know if it's clear now.

Thanks.

**Comment by fabioCollini** [comment]
Hi, you can try to set both module in your rule, the constructor has a varargs argument. Then if you don't override any objects defined in a module it should work (the original objects defined in the module are used).
Let me know if it works (or it will be great if you can reproduce the error in a demo project).

**Comment by danielocampo2** [comment]
I tried that, but since AndroidModule contains some stuff that is 'unmockable', it fails with that error. So as far as I understand from that output, and by following the stacktrace, it is still calling Mockito.mock() in some resource provided by this module instead of just providing the normal resource.
Now I am with something else, but if I have some time I can try to make a demo. But so far that's what is happening. 

Thanks!

**Comment by danielocampo2** [comment]
@fabioCollini Just wanted to let you know that I seemed to have solved this problem but had to modify the library. I am not sure how correct it is my change and even if it'd be useful for other people (I am planning to create a PR at some point).
The solution I implemented is to expose a method in DaggerMockRule to also add modules that should not be mocked and should be built along with the component, by having now 2 lists, the existing modules one, and another one with noMockModules that does the same thing when initialising the component but instead of using the ModuleOverrider it just uses the actual Module class that it is passed.
